modules/tobira/src/main/java/org/opencastproject/tobira/impl/TobiraEndpoint.java

```diff
@@ -204,4 +245,88 @@ private static Response badRequest(String msg) {
     logger.warn("Bad request to tobira/harvest: {}", msg);
     return Response.status(BAD_REQUEST).entity(msg).build();
   }
+
+  @POST
+  @Path("/stats")
+  @Consumes(APPLICATION_JSON)
+  @RestQuery(
+      name = "stats",
+      description = "Accepts a json blob of statistical data about Tobira.",
+      restParameters = {},
+      bodyParameter = @RestParameter(description = "The Tobira data blob",
+            isRequired = true,
+            name = "BODY",
+            type = Type.STRING),
+      responses = {
+          @RestResponse(description = "Stats parsed", responseCode = HttpServletResponse.SC_ACCEPTED)
+      },
+      returnDescription = "No data returned, just a 204 on success"
+  )
+  public Response acceptStats(@Context HttpServletRequest request) {
+    try (InputStream is = request.getInputStream()) {
+      String json = IOUtils.toString(is, request.getCharacterEncoding());
+      stats = gson.fromJson(json, JsonElement.class).getAsJsonObject();
+      stats.addProperty("updated", sdf.format(Calendar.getInstance().getTime()));
+    } catch (IOException e) {
+      logger.error("Error parsing Tobira stats blob", e);
+      return Response.status(BAD_REQUEST).build();
+    }
+    return Response.noContent().build();
+  }
+
+  @GET
+  @Path("/stats")
+  @Produces(APPLICATION_JSON)
+  @RestQuery(name = "stats",
+      description = "Returns the stats, if any, pushed from Tobira",
+      returnDescription = "The stats, or an empty object",
+      responses = {
+          @RestResponse(description = "The stats, or an empty object", responseCode = HttpServletResponse.SC_OK)
+      })
+  public Response getStats() {
+    return Response.ok(gson.toJson(stats)).build();
+  }
+
+  public void setServiceRegistry(ServiceRegistry serviceRegistry) {
+    this.serviceRegistry = serviceRegistry;
+  }
+
+  @Override
+  protected ServiceRegistry getServiceRegistry() {
+    return serviceRegistry;
+  }
+
+  @Reference
+  public void setSecurityService(SecurityService securityService) {
+    this.securityService = securityService;
+  }
+
+  @Override
+  protected SecurityService getSecurityService() {
+    return securityService;
+  }
+
+  @Reference
+  public void setUserDirectoryService(UserDirectoryService userDirectoryService) {
+    this.userDirectoryService = userDirectoryService;
+  }
+  @Override
+  protected UserDirectoryService getUserDirectoryService() {
+    return userDirectoryService;
+  }
+
+  @Reference
+  public void setOrganizationDirectoryService(OrganizationDirectoryService organizationDirectoryService) {
+    this.organizationDirectoryService = organizationDirectoryService;
+  }
+
+  @Override
+  protected OrganizationDirectoryService getOrganizationDirectoryService() {
+    return organizationDirectoryService;
+  }
+
+  @Override
+  protected String process(Job job) throws Exception {
+    throw new RuntimeException("This should never be called for TobiraEndpoint!");
```

Comment:
Deliberately raising a `RuntimeException` to crash the SR in dev if anyone is silly enough to start building Tobira jobs.